ocean_provider/validation/RBAC.py

```diff
@@ -77,6 +77,40 @@ def get_additional_dids(self):
             for additional_input in additional_inputs
         ]
 
+    def get_data(self):
+        if "data" not in self.request.keys():
+            raise Exception("Data to encrypt is empty.")
+        if (
+            not self._check_if_asset()
+            and not self._check_if_respects_file_encryption_schema()
+        ):
+            raise Exception("Invalid type of data.")
+
+        return self.request["data"]
+
+    def _check_if_asset(self):
+        data = self.request["data"]
+        if isinstance(data, dict):
+            if data.get("version") is not None:
+                return True
+        elif isinstance(data, str):
+            data_dict = json.loads(data)
+            if data_dict.get("version") is not None:
+                return True
+        return False
+
+    def _check_if_respects_file_encryption_schema(self):
+        data = self.request["data"]
+        if isinstance(data, list):
```

Comment:
can it be a stringified list?
I don't like the imbrication here, it is not clear to me at first glance what this does. This means you need to improve its readability.

Reply:
@soonhuat is it necessary to check a stringified list of files in provider?

Reply:
I think don't need to the stringified list check
TODAY if provider interface from market or oceanJs, it won't be a stringify list, is always either stringify of DDO object or asset file object as below structure (schema 4.1.0):
{
    nftAddress,
    datatokenAddress,
    files: [
      {
        type: 'url',
        index: 0,
        url: "abcdefg",
        method: 'GET'
      }
    ]
}
based on calina suggestion, maybe rename it to _is_file_encryption_data would able to self explaining itself

Reply:
data is not a list, is an object
and probably line 107 and 108 is enough already, since that's exactly is asset file encryption data structure